### ISSUE TITLE
Vector2.angleRad(Vector2) returns angle toward -y (instead of toward +y)

### ISSUE THREAD

**Issue by jcornaz** [libgdx/libgdx#5385]
#### Issue details
Documentation of `Vector2.angleRad(Vector2)` states:
> return the angle in radians of this vector (point) relative to the given vector. Angles are towards the positive y-axis. (typically counter-clockwise.)

So based on this statement `Vector2.Y.angleRad(Vector2.X)` should return `Math.PI / 2`.
But it returns `-Math.PI / 2`

#### Reproduction steps/code
```java
System.out.println(Vector2.Y.angleRad(Vector2.X)) // prints "-1.5707964" (which is wrong)

// not passing the argument works as expected
System.out.println(Vector2.Y.angleRad()) // prints "1.5707964" (which is correct)
```

#### Version of LibGDX and/or relevant dependencies
1.9.8

#### Please select the affected platforms
Should not be platform related. This is only math. But I observed the problem on Windows.

**Comment by obigu** [comment]
It is actually correct, maybe the wording of the docs could be improved.

Basically the key point here is that order matters.
With an Y positive coordinate system, this method will return the counter clock-wise angle between the current Vector and the reference (argument) Vector. 

If you calculate the counter-clock angle between (0, 1) and (1, 0) you should get 270º which is what
`System.out.println(Vector2.Y.angleRad(Vector2.X))` returns.

On the other hand
`System.out.println(Vector2.X.angleRad(Vector2.Y))` returns 90º (in radians) as expected.

**Comment by jcornaz** [comment]
>It is actually correct, maybe the wording of the docs could be improved.

Sorry I don't understand. How can `Vector2.Y.angleRad(Vector2.X))` returns a different value than `Vector2.Y.angleRad()`?

**Comment by jcornaz** [comment]
By the way, as a side question, may I ask what is the rationale behind the fact that `angle` returns value between `0` and `360` while `angleRad` returns values between `-Math.PI` and `Math.PI`?

Why not return values between `-180` and `180` for `angle` for instance? It would be more consistent and simpler if `angle(x)` would always be equal to `angleRad(x) * MathUtils.radiansToDegrees`.

**Comment by obigu** [comment]
I do agree that the docs can be improved because "relative to" may be misleading. Using your definitions, `Vector2.angleRad(Vector2 argument)`  returns the angle considering the current Vector is the reference. 
So, in your specific example, 
`System.out.println(Vector2.Y.angleRad())` calculates the angle between (0, 1) and (1, 0) which is 90º but `System.out.println(Vector2.Y.angleRad(Vector2.X)) calculates the angle between (1, 0) and (0, 1) which is 270º or -90º.

To your second question I guess the main reason is just that Math.atan2 returns in the range [-Math.pi, +Math.pi] and it's a common convention when working in polar coordinates. It could have been made consistent with degrees but the conversion is trivial and changing it now would break backwards compatibility.

**Comment by jcornaz** [comment]
Let's make it simpler:

And  `Vector2.Y.angleRad(Vector2)` says:

> the angle in radians of this vector (point) relative to the given vector. Angles are towards the positive y-axis. (typically counter-clockwise.)

In the specific case of using the x-axis as a "given vector" like in `Vector2.Y.angleRad(Vector2.X)`. that description could be reworded like this:

> the angle in radians of this vector (point) relative to the **x-axis**. Angles are towards the positive y-axis. (typically counter-clockwise.)

And we can see that's it is now the exact same description as stated in `Vector2.angleRad()`:

> the angle in radians of this vector (point) relative to the **x-axis**. Angles are towards the positive y-axis. (typically counter-clockwise)

From that premise, no matter how one interpret "relative to" and "reference". `v.angleRad()` should always give the exact same result thant `v.angleRad(Vector2.X)`.

**Comment by obigu** [comment]
Agree, the javadoc comment was probably copied from one to another and should be inverted on `Vector2.Y.angleRad(Vector2)`.

**Comment by jcornaz** [comment]
Theses two functions are overloads. They have to behave in a consistent way. So changing the doc is not a solution.

Implementation has to be fixed.

**Comment by mgsx-dev** [comment]
fixed by #5428
